docs/DeveloperGuide.md

```diff
@@ -1,12 +1,251 @@
+  the type of command that needs to be executed.
+* Once validated, a new Command called `ListPassengerCommand` should be instantiated.  
+
+![class diagram](../docs/ug-diagrams/images/passengerListcmdClassDiagram_0.jpg)  
+
+Part 2:  
+
+* `ListPassengerCommand` would then call `execute(passengers, lineInput)`,
+  associating itself with PassengerList to obtain the existing `ArrayList<FlightInfo> passengers`.
+* From there, `listOperation` method would run within method `execute` and
+prints out all `passengers` details on a table format.
+
+![class diagram](../docs/ug-diagrams/images/passengerListcmdClassDiagram_1.jpg)
+
+
+**Sequence diagram**
+
+![sequence diagram](../docs/ug-diagrams/images/passengerListcmdSeqDiagram.jpg)  
+
+When `paser` verifies that the command is an entity `passenger` and `list` operation,
+`ListPassengerCommend` is instantiated.  
+1. `execute(passenger, lineInput)` will run within `ListPassengerCommend`
+   which in turn would call on `listOperation()` in `PassengerList` class.
+2. Within `listOperation()`, it would first check if arraylist `passengers` obtained 
+from `OperationList` is empty or not.
+3. If `passengers` is empty, prints empty table, else prints respective passenger details in table form.
+
+---
+
+### Add a flight feature
+
+The Add a flight function adds a flight with its corresponding details to the flight list.
+
+**Sequence Diagram**
+
+**_NOTE:_** Exceptions are omitted for readability.
+
+![sequence diagram](../docs/ug-diagrams/images/flightAddCmdSeqDiagram.jpg)
+
+When the `Parser` recognizes the `add flight` command, `AddFlightCommand` is instantiated.
+1. The `AddFlightCommand` then implements a new `addOperation(lineInput:String)` in `FlightList`
+2. `FlightList` then instantiates the `FlightInfo` Object using the attributes retrieved from the 
+user input in the previous step.
+3. `Ui` class level method `showFlightAddedMessage()` is used to inform the user they have added a flight.
+
+
+---
+
+### Delete a flight feature
+
+A delete function which allows the user to delete a flight specified with its flight number and
+departure time from the flight list.
+
+This feature is facilitated by `DeleteFlightCommand`. It extends an abstract `Command` with an override
+method `execute`. The abstract `Command` extends a `Parser` which holds and validates the User input
+to determine the type of command.
+
+`DeleteFlightCommand` implements the following operation:
+* execute(OperationList flights, String lineInput) &#8594; An override method inherited from
+`command` to execute the delete operation that should delete the specified flight from the logbook.
+
+The sequence diagram shown below describes the behaviour of the flight delete operation. This scenario
+assumes that the manager has given a valid 'flight delete' command as an input for easier readability.
+
+**Sequence Diagram**
+
+![sequence diagram](../docs/ug-diagrams/images/flightDeleteCmdSeqDiagram.jpg)
```

Comment:
Consider changing the following: 
1. Omit the return from the sequence diagram (the back dotted arrow represents return)
2. Same problems as the previous sequence diagram
3. ui:UI should only be "activated" when called after `showFlightRemovedMessage()`. Therefore the activation bar should only start from the arrow calling the function. 
4. Avoid touching the calling arrow with the previous method. Ths makes the diagram look confusing since the previous callback function did not return and the next function is called. 
![image](https://user-images.githubusercontent.com/39018497/198266572-5f18ca5c-332f-4782-ba7f-01c3abef168b.png)
5. The return (dotted) arrow should be pointing from the bottom of the activation bar of the returning function. (consider elongating some activation bar to fit the arrows)

Reply:
For clarity, do avoid overlaps between the alt condition and the activation bar

<img width="163" alt="2" src="https://user-images.githubusercontent.com/88148155/198495297-cfab7227-25b5-4d20-a799-ff46922858d8.png">